simtfl/node.py

```diff
@@ -78,12 +105,72 @@ def run(self):
         implementation.
         """
         while True:
-            while len(self.mailbox) > 0:
-                (sender, message) = self.mailbox.popleft()
+            while len(self._mailbox) > 0:
+                (sender, message) = self._mailbox.popleft()
                 print(f"T{self.env.now:5d}: handling  {sender:2d} -> {self.ident:2d}: {message}")
                 yield from self.handle(sender, message)
 
             # This naive implementation is fine because we have no actual
             # concurrency.
-            self.wakeup = self.env.event()
-            yield self.wakeup
+            self._wakeup = self.env.event()
+            yield self._wakeup
+
+
+__all__ = ['PassiveNode', 'SequentialNode']
+
+from simpy import Environment
+import unittest
+
+from .message import PayloadMessage
+from .network import Network
+
+
+class PassiveReceiverTestNode(PassiveNode):
+    def __init__(self):
+        super().__init__()
+        self.received = deque()
+
+    def handle(self, sender, message):
+        self.received.append((sender, message, self.env.now))
+        yield self.env.timeout(3)
```

Comment:
I'm confused by the order of the `received` tracking vs the timeout.

If we want to model "a message sent requires 3 ticks before it is received", then shouldn't we swap the order here?

Reply:
Rather than answering my question in this comment, let's update the design docs with a section like "Modeling Network Propagation Delays" where we describe when/where we use `env.timeout` for that purpose, and if we use it for other purposes, we also describe those cases.

Reply:
This isn't modelling a network propagation delay. That is modelled in `Network`.

This is purely adding a delay to message handling so that we can distinguish the behaviour of `SequentialNode` from `PassiveNode` in the test. `SequentialNode` will take 3 seconds between handling each message; `PassiveNode` will not.